src/erlcloud_mon.erl

```diff
@@ -62,8 +62,24 @@ list_metrics(
   DimensionFilter,
   NextToken
  ) ->
+    list_metrics(Namespace, MetricName, DimensionFilter, NextToken, default_config()).
+
+-spec list_metrics(
+        Namespace       ::string(),
+        MetricName      ::string(),
+        DimensionFilter ::[{string(),string()}],
+        NextToken       ::string(),
+        Config          ::aws_config()
+                          ) -> term().
+
+list_metrics(
+  Namespace,
+  MetricName,
+  DimensionFilter,
+  NextToken,
+  Config
```

Comment:
could you please follow other modules in the library ad check Config type either with a guard or pattern match to `#aws_config{}`?